core/bin/dust_api.rs

```diff
@@ -1745,187 +1753,25 @@ async fn data_sources_delete(
-                    })),
-                }),
-            ),
-        },
-    }
-}
-
-#[derive(serde::Deserialize)]
-struct DatabasesListQuery {
-    offset: usize,
-    limit: usize,
-}
-
-async fn databases_list(
-    extract::Path((project_id, data_source_id)): extract::Path<(i64, String)>,
-    extract::Query(query): extract::Query<DatabasesListQuery>,
-    extract::Extension(state): extract::Extension<Arc<APIState>>,
-) -> (StatusCode, Json<APIResponse>) {
-    let project = project::Project::new_from_id(project_id);
-
-    match state
-        .store
-        .list_databases(&project, &data_source_id, Some((query.limit, query.offset)))
-        .await
-    {
-        Err(e) => error_response(
-            StatusCode::INTERNAL_SERVER_ERROR,
-            "internal_server_error",
-            "Failed to list databases",
-            Some(e),
-        ),
-        Ok((dbs, total)) => (
-            StatusCode::OK,
-            Json(APIResponse {
-                error: None,
-                response: Some(json!({
-                    "databases": dbs,
-                    "offset": query.offset,
-                    "limit": query.limit,
-                    "total": total,
-                })),
-            }),
-        ),
-    }
-}
-
-async fn databases_delete(
-    extract::Path((project_id, data_source_id, database_id)): extract::Path<(i64, String, String)>,
-    extract::Extension(state): extract::Extension<Arc<APIState>>,
-) -> (StatusCode, Json<APIResponse>) {
-    let project = project::Project::new_from_id(project_id);
-
-    match state
-        .store
-        .load_database(&project, &data_source_id, &database_id)
-        .await
-    {
-        Err(e) => error_response(
-            StatusCode::INTERNAL_SERVER_ERROR,
-            "internal_server_error",
-            "Failed to retrieve database",
-            Some(e),
-        ),
-        Ok(None) => error_response(
-            StatusCode::NOT_FOUND,
-            "database_not_found",
-            &format!("No database found for id `{}`", database_id),
-            None,
-        ),
-        Ok(Some(db)) => match db.delete(state.store.clone()).await {
-            Err(e) => error_response(
-                StatusCode::INTERNAL_SERVER_ERROR,
-                "internal_server_error",
-                "Failed to delete database",
-                Some(e),
-            ),
-            Ok(_) => (
-                StatusCode::OK,
-                Json(APIResponse {
-                    error: None,
-                    response: Some(json!({
-                        "success": true
-                    })),
-                }),
-            ),
-        },
-    }
-}
-
 #[derive(serde::Deserialize)]
 struct DatabasesTablesUpsertPayload {
     table_id: String,
     name: String,
     description: String,
 }
 
-async fn databases_tables_upsert(
-    extract::Path((project_id, data_source_id, database_id)): extract::Path<(i64, String, String)>,
+async fn tables_upsert(
```

Comment:
every table-related endpoint:
- is renamed from `databases_tables_*` inot `tables_*`
- drops the `database_id` PATH parameter in the url